TexasHoldem.c

```diff
@@ -1,7 +1,456 @@
+		{
+			correct = 1;
+		}
+		else
+		{
+			correct = 0;
+		}
+		break;
+	}
+	return(correct);
+}
+
+//This function draws the community cards
+void randomCommunityCard(int draw, int communitycards[][2])
+{
+	int r, face, value, face2, value2, face3, value3;
+	r = rand();
+	face = (r% 4)+1;
+	value = (r% 13) +1;
+	switch(draw)
+	{
+		case 1: //The Flop
+		
+		do {
+			face2 = (r% 4)+1;
+			value2 = (r% 13) +1;
+		} while (face == face2 && value == value2);
+		
+		do {
+			face3 = (r% 4)+1;
+			value3 = (r% 13) +1;
+		} while ( (face == face3 && value == value3) || (face2 == face3 && value2 == value3) );
+	
+		communitycards[0][0] = face;
+		communitycards[0][1] = value;
+		communitycards[1][0] = face2;
+		communitycards[1][1] = value2;
+		communitycards[2][0] = face3;
+		communitycards[2][1] = value3;
+		break;
+		
+		case 2: //The Turn
+		communitycards[3][0] = face;
+		communitycards[3][1] = value;
+		break;
+		
+		case 3: //The River
+		communitycards[4][0] = face;
+		communitycards[4][1] = value;
+		break;
+	}
+}
+
+//this function checks to make sure the cards/card chosen by randomCommunityCard is not already chosen
+int CheckRandCommunity(int draw, int communitycards[][2])
+{
+	int correct = 0;
+	switch(draw)
+	{
+		case 1: //The Flop
+		{
+			for (i=0; i<3; i++)
+			{
+				for (j=0; j<2; j++)
+				{
+					if (communitycards[i][0] == player1cards[j][0] && communitycards[i][1] == player1cards[j][1])
+						correct = 1;
+					if (communitycards[i][0] == player2cards[j][0] && communitycards[i][1] == player2cards[j][1])
+						correct = 1;
+					if (communitycards[i][0] == player3cards[j][0] && communitycards[i][1] == player3cards[j][1])
+						correct = 1;
+					if (communitycards[i][0] == player4cards[j][0] && communitycards[i][1] == player4cards[j][1])
+						correct = 1;
+				}
+			}
+		}
+		break;
+
+		case 2: //The Turn
+			for (j=0; j<2; j++)
+			{
+				if (communitycards[3][0] == player1cards[j][0] && communitycards[3][1] == player1cards[j][1])
+					correct = 1;
+				if (communitycards[3][0] == player2cards[j][0] && communitycards[3][1] == player2cards[j][1])
+					correct = 1;
+				if (communitycards[3][0] == player3cards[j][0] && communitycards[3][1] == player3cards[j][1])
+					correct = 1;
+				if (communitycards[3][0] == player4cards[j][0] && communitycards[3][1] == player4cards[j][1])
+					correct = 1;
+			}
+		break;
 
+		case 3: //The River
+			for (j=0; j<2; j++)
+			{
+				if (communitycards[4][0] == player1cards[j][0] && communitycards[4][1] == player1cards[j][1])
+					correct = 1;
+				if (communitycards[4][0] == player2cards[j][0] && communitycards[4][1] == player2cards[j][1])
+					correct = 1;
+				if (communitycards[4][0] == player3cards[j][0] && communitycards[4][1] == player3cards[j][1])
+					correct = 1;
+				if (communitycards[4][0] == player4cards[j][0] && communitycards[4][1] == player4cards[j][1])
+					correct = 1;
+				}
+			}
+		break;
+	}
+	return(correct);
```

Comment:
The same comments i have above apply here

Reply:
there is also an extra end bracket in the function. I removed the one right before the last break.

Reply:
correction: i didn't remove it but it should work if you do.